USAGOV-1292-keep-ssg-disabled-state-between-deployments: Running checks

arpage opened the pull request:
<!--- Provide a general summary of your changes in the title above -->
## Jira Task
<!--- Provide a link to the Jira ticket -->
https://cm-jira.usa.gov/browse/USAGOV-1292

## Description
<!--- Summarize the changes made in this pull request, not what it's for. -->
Keep track of tome-disabled state, and leave it where the script(s) found it, if possible.
## Type of Changes
<!--- Put an `x` in all the boxes that apply. -->
- [ ] New Feature
- [ ] Bugfix
- [ ] Frontend (Twig, Sass, JS)
  - Add screenshot showing what it should look like
- [ ] Drupal Config (requires "drush cim")
- [ ] New Modules (requires rebuild)
- [ ] Infrastructure
  - [ ] CMS
  - [ ] WAF
  - [ ] Egress
  - [ ] Tools
- [ ] Other

## Testing Instructions
<!-- This instructions are different from “testing instructions” in Jira – those are typically for Content/UX stakeholders -->
<!-- Not “see Jira” – if they are really the same, copy and paste. -->

### Requires New Config
- [ ] Yes
- [ ] No

### Requires New Content
- [ ] Yes
- [ ] No

### Validation Steps
- Test instruction 1
- Test instruction 2
- Test instruction 3

## Security Review
<!-- Checkboxes to indicate need for review -->

- [ ] Adds/updates software (including a library or Drupal module)
- [ ] Communication with external service
- [ ] Changes permissions or workflow
- [ ] Requires SSPP updates


## Reviewer Reminders
- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions
Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.

arpage committed:
USAGOV-1292-keep-ssg-disabled-state-between-deployments: Running checks

akf commented on bin/snapshot-backups/local-snapshot-backup:
```diff
@@ -63,12 +83,18 @@ if [ "$RETVAL" = "0" ]; then
-  if [ x$echo = x ]; then
-    bin/deploy/try-tome-enable
-  fi
+  #if [ x$echo = x ]; then
```
Did you mean to comment out this if $echo?

akf commented on bin/snapshot-backups/local-snapshot-backup:
```diff
@@ -50,6 +61,15 @@ shift
 #shift
 
+echo "[$TOME_DISABLED_STATE]"
+if [ x$TOME_DISABLED_STATE != 1 ]; then
```
The different tests against $TOME_DISABLED_STATE are catching my eye. I'm pretty sure this one will always evaluate to true: `[ x$TOME_DISABLED_STATE != 1 ]`

akf commented on bin/snapshot-backups/local-snapshot-backup:
```diff
@@ -63,12 +83,18 @@ if [ "$RETVAL" = "0" ]; then
-    bin/deploy/try-tome-enable
-  fi
+  #if [ x$echo = x ]; then
+    if [ x$TOME_DISABLED_STATE != 1 ]; then
```
Again, this looks like a typo.

akf commented on bin/snapshot-backups/downsync-cms:
```diff
@@ -82,7 +98,12 @@ echo  aws s3 sync --only-show-errors $WEB_FILES s3://$S3_BUCKET/cms/public/ --ac
 echo cf ssh cms -c ". /etc/profile; drush cr; drush updatedb --no-cache-clear -y; drush cim -y || drush cim -y; drush cim -y; drush php-eval 'node_access_rebuild();' -y"
 $echo cf ssh cms -c ". /etc/profile; drush cr; drush updatedb --no-cache-clear -y; drush cim -y || drush cim -y; drush cim -y; drush php-eval 'node_access_rebuild();' -y"
 
+if [ x$TOME_DISABLED_STATE = x0 ]; then
```
It turns out we're deleting the usagov.tome_run_disabled variable rather than setting it to 0 (which is probably why the other tests in this changeset check for 1). 
So maybe what's wanted here is 
`if [x$TOME_DISABLED_STATE != x1 ]` ?

akf commented on bin/snapshot-backups/downsync-cms:
```diff
@@ -22,12 +22,28 @@ then
   shift
 fi
 
+needsDisabled=0
```
I'd like this variable name to include "spaceSSH" or just "SSH". `needsSpaceSSHDisabled`

akf reviewed (changes requested):
I see some issues with the TOME_DISABLED_STATE tests.

arpage committed:
Merge branch 'dev' into USAGOV-1292-keep-ssg-disabled-state-between-deployments

arpage committed:
Respond to comments

arpage committed:
Respond to comments

arpage committed:
Merge remote-tracking branch 'origin/dev' into USAGOV-1292-keep-ssg-disabled-state-between-deployments

arpage committed:
final cleanup

arpage commented:
@akf - I've addressed your concerns, and tested local-snapshot-backup in dev, and the tome disabled state is retained after the backup is complete.

As the acid test, I ran the version from dev and tome is always re-enabled after local-snapshot-backup has completed.  

So, these changes seem to fix the issue we are currently experiencing.

arpage committed:
Merge branch 'dev' into USAGOV-1292-keep-ssg-disabled-state-between-deployments

arpage committed:
Merge branch 'dev' into USAGOV-1292-keep-ssg-disabled-state-between-deployments

akf commented on bin/snapshot-backups/downsync-cms:
```diff
@@ -39,13 +39,13 @@ if [ x$1 == x"--dryrun" ]; then
       assertSpaceSSHEnabled $startspace
   fi
-  needsDisabled=1
+  needsSpacedSSHDisabled=1
```
Typo! "d" after "Space"

arpage replied:
I fixed this :)

akf committed:
Merge branch 'dev' into USAGOV-1292-keep-ssg-disabled-state-between-deployments

arpage committed:
Merge branch 'dev' into USAGOV-1292-keep-ssg-disabled-state-between-deployments

arpage committed:
USAGOV-1292-keep-ssg-disabled-state-between-deployments: fix logic nesting and env var typo

arpage committed:
USAGOV-1292-keep-ssg-disabled-state-between-deployments: fix env var typo

akf reviewed (approved):
LGTM
